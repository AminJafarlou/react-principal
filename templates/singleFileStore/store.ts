import { createReducer } from "react-principal";

const SET_NAME = Symbol();

interface InitialState {
  name: string;
}

const initialState: InitialState = {
  name: "",
};

const reducer = createReducer<InitialState>({
  [SET_NAME]: (state, { payload: { name } }) => ({
    ...state,
    name,
  }),
});

function setName(name: string) {
  return { type: SET_NAME, payload: { name } };
}

export { reducer, initialState, setName };
