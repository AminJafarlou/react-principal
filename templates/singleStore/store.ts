import { createReducer, createStore } from "react-principal";

const LOGIN = Symbol();
const LOGOUT = Symbol();

interface InitialState {
  accessToken?: string;
}

const initialState: InitialState = {
  accessToken: undefined,
};

const reducer = createReducer<InitialState>({
  [LOGIN]: (state, { payload: { accessToken } }) => ({
    ...state,
    accessToken,
  }),
  [LOGOUT]: () => ({
    ...initialState,
  }),
});

const AuthStore = createStore({
  reducer,
  initialState,
  persistKey: "AUTHENTICATION",
});

export { reducer, AuthStore };
