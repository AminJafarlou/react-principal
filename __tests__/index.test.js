import renderer, { act } from "react-test-renderer";
import React from "react";
import { ExampleStoreTheme, changeTheme } from "../examples/store";
import { Provider } from "../lib";

function TextInputTester() {
  const { theme } = ExampleStoreTheme.useState();
  const dispatch = ExampleStoreTheme.useDispatch();

  const toggleTheme = () => {
    dispatch(changeTheme(theme === "light" ? "dark" : "light"));
  };

  return (
    <p theme={theme} toggleTheme={toggleTheme}>
      {theme}
    </p>
  );
}

let component;
let tree;
test("store: create theme store", () => {
  component = renderer.create(
    <Provider store={ExampleStoreTheme}>
      <TextInputTester />
    </Provider>,
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("store: toggle theme", () => {
  act(() => {
    tree.props.toggleTheme();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
