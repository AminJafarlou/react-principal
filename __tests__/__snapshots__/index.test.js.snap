// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store: create theme store 1`] = `
<p
  theme="light"
  toggleTheme={[Function]}
  wrong={[Function]}
>
  light
</p>
`;

exports[`store: toggle theme 1`] = `
<p
  theme="dark"
  toggleTheme={[Function]}
  wrong={[Function]}
>
  dark
</p>
`;
